<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1111</width>
    <height>880</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="layoutDirection">
   <enum>Qt::LeftToRight</enum>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>1111</width>
     <height>880</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>1111</width>
     <height>880</height>
    </size>
   </property>
   <property name="baseSize">
    <size>
     <width>1111</width>
     <height>880</height>
    </size>
   </property>
   <property name="focusPolicy">
    <enum>Qt::TabFocus</enum>
   </property>
   <property name="contextMenuPolicy">
    <enum>Qt::NoContextMenu</enum>
   </property>
   <property name="layoutDirection">
    <enum>Qt::LeftToRight</enum>
   </property>
   <property name="autoFillBackground">
    <bool>false</bool>
   </property>
   <widget class="QWidget" name="port_conn" native="true">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>1071</width>
      <height>141</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
      <horstretch>0</horstretch>
      <verstretch>2</verstretch>
     </sizepolicy>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">#port_conn {
   border: 2px solid #808080;
   border-radius: 1px;

}</string>
    </property>
    <widget class="QLabel" name="cam_promt_label">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>50</y>
       <width>141</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Cam Port</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QComboBox" name="com_select">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>50</y>
       <width>151</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
    </widget>
    <widget class="QLabel" name="baud_label">
     <property name="geometry">
      <rect>
       <x>350</x>
       <y>50</y>
       <width>150</width>
       <height>40</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Baud Rate</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QPushButton" name="open_port">
     <property name="geometry">
      <rect>
       <x>660</x>
       <y>50</y>
       <width>121</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Open</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="port_status">
     <property name="geometry">
      <rect>
       <x>790</x>
       <y>50</y>
       <width>140</width>
       <height>40</height>
      </rect>
     </property>
     <property name="minimumSize">
      <size>
       <width>140</width>
       <height>0</height>
      </size>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="styleSheet">
      <string notr="true">#port_status {
   color: red;
   background-color: black;
   padding: 5px;
   border: none;
   border-radius: 0;
   outline: none;
}</string>
     </property>
     <property name="text">
      <string>OFFLINE</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="refresh_port">
     <property name="geometry">
      <rect>
       <x>940</x>
       <y>50</y>
       <width>121</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Refresh</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="baud_input">
     <property name="geometry">
      <rect>
       <x>490</x>
       <y>50</y>
       <width>140</width>
       <height>40</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="text">
      <string>115200</string>
     </property>
     <property name="maxLength">
      <number>10</number>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="reg_edit" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>180</y>
      <width>521</width>
      <height>511</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">#reg_edit {
   border: 2px solid #808080;
   border-radius: 0px;
   padding: 10px;
   margin: 10px;
}</string>
    </property>
    <widget class="QLabel" name="reg_edit_title">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>20</y>
       <width>451</width>
       <height>51</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>20</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Register Edit</string>
     </property>
    </widget>
    <widget class="QLabel" name="cmd_label">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>80</y>
       <width>191</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="text">
      <string>CMD</string>
     </property>
    </widget>
    <widget class="QComboBox" name="cmd_select">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>80</y>
       <width>260</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
      </font>
     </property>
     <item>
      <property name="text">
       <string>1 Byte Write</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>2 Byte Write</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>3 Byte Write</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>4 Byte Write</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Read</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="addr_label">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>130</y>
       <width>191</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Address</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="addr_inp">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>130</y>
       <width>260</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="inputMask">
      <string/>
     </property>
     <property name="maxLength">
      <number>4</number>
     </property>
    </widget>
    <widget class="QLabel" name="reg_data_label">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>180</y>
       <width>191</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Data</string>
     </property>
    </widget>
    <widget class="QLabel" name="cmd_to_send_label">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>230</y>
       <width>191</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="text">
      <string>CMD Hex</string>
     </property>
    </widget>
    <widget class="QLabel" name="res_label">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>390</y>
       <width>141</width>
       <height>40</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Response</string>
     </property>
    </widget>
    <widget class="QLabel" name="reg_data_label_2">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>440</y>
       <width>141</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Reg Data</string>
     </property>
    </widget>
    <widget class="QLabel" name="cmd_hex_out">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>230</y>
       <width>260</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="res_out">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>390</y>
       <width>171</width>
       <height>40</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>-1</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="styleSheet">
      <string notr="true">#res_out {
    color: #888888;
    font-size: 14px;
    font-weight: bold;
}
</string>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="reg_data_out">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>440</y>
       <width>171</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="styleSheet">
      <string notr="true">reg_data_out {
    color: #888888;
    font-size: 14px;
    font-weight: bold;
}
</string>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QPushButton" name="execute_cmd">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>320</y>
       <width>461</width>
       <height>51</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Execute</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="data_inp">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>180</y>
       <width>140</width>
       <height>41</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="text">
      <string>00000000</string>
     </property>
     <property name="maxLength">
      <number>8</number>
     </property>
     <property name="dragEnabled">
      <bool>true</bool>
     </property>
     <property name="clearButtonEnabled">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="reset">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>430</y>
       <width>161</width>
       <height>61</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>RESET Cam</string>
     </property>
    </widget>
    <widget class="QPushButton" name="plus_data">
     <property name="geometry">
      <rect>
       <x>370</x>
       <y>180</y>
       <width>30</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>+</string>
     </property>
    </widget>
    <widget class="QPushButton" name="min_data">
     <property name="geometry">
      <rect>
       <x>410</x>
       <y>180</y>
       <width>30</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>-</string>
     </property>
    </widget>
    <widget class="QPushButton" name="clr_data">
     <property name="geometry">
      <rect>
       <x>450</x>
       <y>180</y>
       <width>30</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>X</string>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="db_box" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>720</y>
      <width>1091</width>
      <height>151</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">#db_box {
   border: 2px solid #808080;
   border-radius: 0px;
   padding: 10px;
   margin: 10px;
}</string>
    </property>
    <widget class="QLabel" name="db_select_label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>251</width>
       <height>131</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>20</pointsize>
      </font>
     </property>
     <property name="styleSheet">
      <string notr="true"/>
     </property>
     <property name="text">
      <string>Select Table</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QComboBox" name="db_select_inp">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>40</y>
       <width>431</width>
       <height>71</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>20</pointsize>
      </font>
     </property>
     <property name="currentIndex">
      <number>-1</number>
     </property>
     <property name="maxVisibleItems">
      <number>60</number>
     </property>
    </widget>
    <widget class="QPushButton" name="db_export">
     <property name="geometry">
      <rect>
       <x>750</x>
       <y>80</y>
       <width>161</width>
       <height>51</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Export Single</string>
     </property>
    </widget>
    <widget class="QPushButton" name="db_import">
     <property name="geometry">
      <rect>
       <x>750</x>
       <y>20</y>
       <width>161</width>
       <height>51</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Import Single</string>
     </property>
    </widget>
    <widget class="QPushButton" name="importAll">
     <property name="geometry">
      <rect>
       <x>920</x>
       <y>20</y>
       <width>141</width>
       <height>51</height>
      </rect>
     </property>
     <property name="text">
      <string>Import All</string>
     </property>
    </widget>
    <widget class="QPushButton" name="exportAll">
     <property name="geometry">
      <rect>
       <x>920</x>
       <y>80</y>
       <width>141</width>
       <height>51</height>
      </rect>
     </property>
     <property name="text">
      <string>Export All</string>
     </property>
    </widget>
   </widget>
   <widget class="QComboBox" name="table_select_inp">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>630</y>
      <width>180</width>
      <height>50</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QPushButton" name="updateTable">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>630</y>
      <width>180</width>
      <height>50</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Update THIS Table</string>
    </property>
   </widget>
   <widget class="QPushButton" name="loadTable">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>630</y>
      <width>80</width>
      <height>50</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Load</string>
    </property>
   </widget>
   <widget class="QPushButton" name="tableExe">
    <property name="geometry">
     <rect>
      <x>1010</x>
      <y>630</y>
      <width>80</width>
      <height>50</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Execute</string>
    </property>
   </widget>
   <widget class="QTableWidget" name="tableWidget">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>230</y>
      <width>551</width>
      <height>391</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="readAll">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>690</y>
      <width>181</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Read Cam Settings</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="searchBar">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>190</y>
      <width>551</width>
      <height>30</height>
     </rect>
    </property>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
